src/database/migrations/postgres/20240328152300_task_events_table.sql

```diff
@@ -0,0 +1,14 @@
+CREATE SEQUENCE task_id_seq AS BIGINT;
```

Comment:
I think in near future we will need to have `Task` as a database entity anyway, i.e. to store aggregate version, maybe last snapshot ref. That would make both Postgres and MySQL require a dedicated "tasks" table with it's own auto-incremented ID.

Reply:
I don't see a need on today to generate app-level identifiers. If there will be a need, they would be UUIDs, not numbers. Right now, sequence on the database side is a good match, even if we add multiple API server instances. Regarding MySQL, the solution above with a separate `tasks` table and auto-increment ID will be just fine, and it does not have to match Postgres schema bit by bit.

Reply:
In my prev. projects we only used application-side identity as it is very nice for sharding - you don't need any central point for getting IDs and can easily split objects across databases. Which we'll likely have to do for the public node data volumes.

It would also be interesting to see a concurrent transaction test. If two concurrent transactions are creating a new Task - can incrementing a sequence result in OCC error?

Reply:
From concurrency point of view, it is safe. Sequences are managed resource, which is independent on application transactions. There is no determinism in values N concurrent transactions would get, but the values themselves are guaranteed to be unique. In case of other conflict, some sequence values are skipped, they are not reversible, reusable, so new attempts will simply take next ticks.

https://www.postgresql.org/docs/current/functions-sequence.html

```
nextval ( regclass ) → bigint

Advances the sequence object to its next value and returns that value. This is done atomically: even if multiple sessions execute nextval concurrently, each will safely receive a distinct sequence value. If the sequence object has been created with default parameters, successive nextval calls will return successive values beginning with 1. Other behaviors can be obtained by using appropriate parameters in the [CREATE SEQUENCE](https://www.postgresql.org/docs/current/sql-createsequence.html) command.

```


https://dev.mysql.com/doc/refman/8.0/en/innodb-auto-increment-handling.html


SQLite is not intented for concurrent access in general, so it's not an issue there.

Reply:
For future - I do not mind to switch to application level identifiers, as long as we switch the data type from `i64` to `uuid`.